### PR TITLE
[fa] Convolutional neural networks

### PR DESCRIPTION
**Pull request by AlisterTA**
A new PR for Convolutional Neural Networks. The old one is available [here](https://github.com/shervinea/cheatsheet-translation/pull/127).

highlighting a few side-notes @erfannoury 

Inception Network : @ehsanmk translated it as `ابتکاری` but to me it doesn’t seem to be accurate. The name `Inception` actually comes from the movie "Inception", and the model is inspired from the meme “We need to go Deeper". In Persian it’s translated to `تلقین ` but I personally would stick with the original word. Other translation did the same. what's you take on this?

Intersection over Union: I have a few suggestions: `نسبت هم‌پوشانی اشتراک به اجتماع `
or `نسبت همبری به اجتماع`
Non-max suppression:`فروداشت غیربیشینه`
Padding: IMO `لایه گذاری` does not convey the meaning of the original term very accurately, how about ` گسترش مرز` ?

`one-shot` :how about `یک‌باره‌ای`? And I assume it is important to include the original term in parentheses, because often a translation does not convey the shades of meanings contained in the original term.

People from academia usually translate `generative adversarial networks` to `شبکه های مولد تخاصمی` but to me `تخاصمی` is more like `hostile` than `adversarial`. `تخاصمی` is also a non-Persian word.( don't mean to be picky or anything). i'd like to offer a new term `هماورد`  or `هم‌آورد`. what's your take on this?

You might not find my translation quite fluent and accurate and there are definitely room for improvement.  I'm looking forward to seeing your review and suggestions

### CODE REVIEW

File: fa/convolutional-neural-networks.md [hunk 1]
```diff
@@ -4,713 +4,919 @@
...
+</div>
 
 <br>
 
 
 **33. Parameter compatibility in convolution layer ― By noting I the length of the input volume size, F the length of the filter, P the amount of zero padding, S the stride, then the output size O of the feature map along that dimension is given by:**
 
-&#10230;
+<div dir="rtl">
+سازش‌پذیری فراسنج در لایه کانولوشنی – با ذکر I به عنوان طول اندازه توده ورودی،F طول فیلتر، P میزان گسترش مرزِ صفر، S گام، اندازه خروجی نگاشت ویژگی O در امتداد ابعاد مفروض است:
```
Comment (erfannoury) [inline fa/convolutional-neural-networks.md]: مفروض است -> خواهد بود

File: fa/convolutional-neural-networks.md [hunk 2]
```diff
@@ -4,713 +4,919 @@
...
+</div>
 
 <br>
 
 
 **38. [One bias parameter per filter, In most cases, S<F, A common choice for K is 2C]**
 
-&#10230;
+<div dir="rtl">
+[یک پیش‌قدر به ازای هر فیلتر، در بیشتر موارد S&lt;F است، یک انتخاب رایج برای K، 2C است]
```
Comment (erfannoury) [inline fa/convolutional-neural-networks.md]: Are you going with پیش‌قدر for bias? I think I have no problem with it. I should probably update the `persian-ml-glossary` soon.

File: fa/convolutional-neural-networks.md [hunk 3]
```diff
@@ -4,713 +4,919 @@
...
+
 
 <br>
 
 
 **39. [Pooling operation done channel-wise, In most cases, S=F]**
 
-&#10230;
+<div dir="rtl">
+[عملیات ادغام به کانال‌به‌کانال انجام میشود، در بیشتر موارد S=F است]
```
Comment (erfannoury) [inline fa/convolutional-neural-networks.md]: به صورت کانال‌به‌کانال

File: fa/convolutional-neural-networks.md [hunk 4]
```diff
@@ -4,713 +4,919 @@
...
+</div>
 
 <br>
 
-
 **40. [Input is flattened, One bias parameter per neuron, The number of FC neurons is free of structural constraints]**
 
-&#10230;
+<div dir="rtl">
+[ورودی مسطح شده است، یک پیش‌قدر به ازای هر نورون، تعداد نرون‌های FC فاقد محدودیت‌های ساختاری‌ست]
```
Comment (erfannoury) [inline fa/convolutional-neural-networks.md]: نرون ـ> نورون

File: fa/convolutional-neural-networks.md [hunk 5]
```diff
@@ -4,713 +4,919 @@
...
+</div>
 
 <br>
 
 
 **41. Receptive field ― The receptive field at layer k is the area denoted Rk×Rk of the input that each pixel of the k-th activation map can 'see'. By calling Fj the filter size of layer j and Si the stride value of layer i and with the convention S0=1, the receptive field at layer k can be computed with the formula:**
 
-&#10230;
+<div dir="rtl">
+ناحیه تاثیر – ناحیه تاثیر در لایه k محدودهای از ورودی Rk×Rk است که هر پیکسلِ kاٌم نگاشت ویژگی می‌تواند 'ببیند'. با ذکر Fj به عنوان اندازه فیلتر لایه j و Si مقدار گام لایه i و با این توافق که S0=1 است، ناحیه تاثیر در لایه k با فرمول زیر محاسبه می‌شود:
```
Comment (erfannoury) [inline fa/convolutional-neural-networks.md]: محدودهای ->‌ محدوده‌ای

File: fa/convolutional-neural-networks.md [hunk 6]
```diff
@@ -4,713 +4,919 @@
...
+</div>
 
 <br>
 
 
 **44. Rectified Linear Unit ― The rectified linear unit layer (ReLU) is an activation function g that is used on all elements of the volume. It aims at introducing non-linearities to the network. Its variants are summarized in the table below:**
 
-&#10230;
+<div dir="rtl">
+(ReLU) یک تابع فعال‌سازی g است که بر روی تمامی عناصر توده اعمال میشود. هدف آن ارائه (رفتار) غیرخطی به شبکه است. انواع آن در جدول زیر بصورت خلاصه آمده‌اند:
```
Comment (erfannoury) [inline fa/convolutional-neural-networks.md]: No translation for the header (Rectified Linear Unit -> تابع همسوساز خطی)

File: fa/convolutional-neural-networks.md [hunk 7]
```diff
@@ -4,713 +4,919 @@
...
+</div>
 
 <br>
 
 
 **47. Softmax ― The softmax step can be seen as a generalized logistic function that takes as input a vector of scores x∈Rn and outputs a vector of output probability p∈Rn through a softmax function at the end of the architecture. It is defined as follows:**
 
-&#10230;
+<div dir="rtl">
+x∈Rn را از ورودی می‌گیرد و یک بردار خروجی احتمال p∈Rn، به‌واسطه‌ی تابع بیشینه‌ی هموار در انتهای معماری، تولید می‌کند. این تابع بصورت زیر تعریف می‌شود:
```
Comment (erfannoury) [inline fa/convolutional-neural-networks.md]: First line has not been translated.

File: fa/convolutional-neural-networks.md [hunk 8]
```diff
@@ -4,713 +4,919 @@
...
+</div>
 
 <br>
 
 
 **50. Types of models ― There are 3 main types of object recognition algorithms, for which the nature of what is predicted is different. They are described in the table below:**
 
-&#10230;
+<div dir="rtl">
+انواع مدل‌ها – سه نوع اصلی از الگوریتم‌های بازشناسایی وجود دارد، که ماهیت آنچه‌که شناسایی شده متفاوت است. این الگوریتم‌ها در جدول زیر توضیح داده شده‌اند:
```
Comment (erfannoury) [inline fa/convolutional-neural-networks.md]: انواع مدل~ها~

File: fa/convolutional-neural-networks.md [hunk 9]
```diff
@@ -4,713 +4,919 @@
...
+</div>
 
 <br>
 
 
 **53. [Classifies a picture, Predicts probability of object, Detects an object in a picture, Predicts probability of object and where it is located, Detects up to several objects in a picture, Predicts probabilities of objects and where they are located]**
 
-&#10230;
+<div dir="rtl">
+[یک عکس را دستهبندی میکند، احتمال شیء را پیشبینی میکند، یک شیء را در یک عکس شناسایی میکند، احتمال یک شیء و موقعیت آن را پیشبینی میکند، چندین شیء در یک عکس را شناسایی میکند، احتمال اشیاء و موقعیت آنها را پیشبینی میکند]
```
Comment (erfannoury) [inline fa/convolutional-neural-networks.md]: دسته‌بندی

File: fa/convolutional-neural-networks.md [hunk 10]
```diff
@@ -4,713 +4,919 @@
...
+</div>
 
 <br>
 
 
 **54. [Traditional CNN, Simplified YOLO, R-CNN, YOLO, R-CNN]**
 
-&#10230;
+<div dir="rtl">
+[CNN سنتی، شکل ساده YOLO، R-CNN، YOLO، R-CNN]
```
Comment (erfannoury) [inline fa/convolutional-neural-networks.md]: Simplified YOLO:
<div dir="rtl">
YOLO ساده شده
</div>

File: fa/convolutional-neural-networks.md [hunk 11]
```diff
@@ -4,713 +4,919 @@
...
+</div>
 
 <br>
 
 
 **56. [Bounding box detection, Landmark detection]**
 
-&#10230;
+<div dir="rtl">
+[پیش‌بینی کادر محصورکننده، ]شناسایی نقاط(برجستگی)
```
Comment (erfannoury) [inline fa/convolutional-neural-networks.md]: برجستگی ->‌ برجسته
Move `[` to the end

File: fa/convolutional-neural-networks.md [hunk 12]
```diff
@@ -4,713 +4,919 @@
...
+</div>
 
 <br>
 
 
 **57. [Detects the part of the image where the object is located, Detects a shape or characteristics of an object (e.g. eyes), More granular]**
 
-&#10230;
+<div dir="rtl">
+[بخشی از تصویر که شیء قرارگرفته را شناسایی می‌کند، یک شکل یا مشخصات یک شیء (مثل چشم‌ها) را شناسایی می‌کند، موشکافانه‌تر]
```
Comment (erfannoury) [inline fa/convolutional-neural-networks.md]: که شی **در آن** قرار گرفته

File: fa/convolutional-neural-networks.md [hunk 13]
```diff
@@ -4,713 +4,919 @@
...
+</div>
 
 <br>
 
 
 **61. Anchor boxes ― Anchor boxing is a technique used to predict overlapping bounding boxes. In practice, the network is allowed to predict more than one box simultaneously, where each box prediction is constrained to have a given set of geometrical properties. For instance, the first prediction can potentially be a rectangular box of a given form, while the second will be another rectangular box of a different geometrical form.**
 
-&#10230;
+<div dir="rtl">
+کادرهای محور – کادر بندی محور روشی است که برای پیشبینی کادرهای محصورکننده هم‌پوشان استفاده می‌شود. در عمل، شبکه این اجازه را دارد که بیش از یک کادر بصورت هم‌زمان پیشبینی کند جایی‌که هر پیش‌بینی کادر مقید به داشتن یک مجموعه خصوصیات هندسی مفروض است. به عنوان مثال، اولین پیش‌بینی می‌تواند یک کادر مستطیلی با قالب خاص باشد حال آنکه کادر دوم، یک کادر مستطیلی محور با قالب هندسی متفاوتی خواهد بود.
```
Comment (erfannoury) [inline fa/convolutional-neural-networks.md]: محور ->‌ محوری

File: fa/convolutional-neural-networks.md [hunk 14]
```diff
@@ -4,713 +4,919 @@
...
+</div>
 
 <br>
 
 
 **62. Non-max suppression ― The non-max suppression technique aims at removing duplicate overlapping bounding boxes of a same object by selecting the most representative ones. After having removed all boxes having a probability prediction lower than 0.6, the following steps are repeated while there are boxes remaining:**
 
-&#10230;
+<div dir="rtl">
+فروداشت غیربیشینه – هدف روش فروداشت غیربیشبنه، حذف کادرهای مصورکننده هم‌پوشان تکراریِ دسته یکسان با انتخاب معرف‌ترین ها است. بعد از حذف همه کاردهایی که احتمال پیش‌بینی پاییتر از 0.6 دارند، مراحل زیر  با وجود آنکه کادرهایی باقی می‌مانند تکرار میشوند:
```
Comment (erfannoury) [inline fa/convolutional-neural-networks.md]: غیربیشبنه ->‌ غیربیشینه
معرف‌ترین‌ها
کاردهایی -> کادرهایی

File: fa/convolutional-neural-networks.md [hunk 15]
```diff
@@ -4,713 +4,919 @@
...
+</div>
 
 <br>
 
 
 **65. YOLO ― You Only Look Once (YOLO) is an object detection algorithm that performs the following steps:**
 
-&#10230;
+<div dir="rtl">
+YOLO – شما فقط یک‌بار نگاه می‌کنید (YOLO) یک الگوریتم شناسایی شیء که مراحل زیر را اجرا میکند:
```
Comment (erfannoury) [inline fa/convolutional-neural-networks.md]: «شما فقط یک بار نگاه می‌کنید»
شناسایی شی **است** که ...

File: fa/convolutional-neural-networks.md [hunk 16]
```diff
@@ -4,713 +4,919 @@
...
+</div>
 
 <br>
 
 
 **67. where pc is the probability of detecting an object, bx,by,bh,bw are the properties of the detected bouding box, c1,...,cp is a one-hot representation of which of the p classes were detected, and k is the number of anchor boxes.**
 
-&#10230;
+<div dir="rtl">
+-----
```
Comment (erfannoury) [inline fa/convolutional-neural-networks.md]: <div dir="rtl">
که pc احتمال شناسایی یک شی است، bx، by، bh، bw اندازه‌های نسبی کادر محیطی شناسایی شده است، c1, ..., cp نمایش «تک‌فعال» یک دسته از p دسته که تشخیص داده شده است، و k تعداد کادرهای محوری است.
</div>

File: fa/convolutional-neural-networks.md [hunk 17]
```diff
@@ -4,713 +4,919 @@
...
+</div>
 
 <br>
 
 
 **68. Step 3: Run the non-max suppression algorithm to remove any potential duplicate overlapping bounding boxes.**
 
-&#10230;
+<div dir="rtl">
+گام سوم: الگوریتم فروداشت غیریبیشینه را برای حذف هر کادر مصورکننده هم‌پوشان تکراری بالقوه، اجرا کن.
```
Comment (erfannoury) [inline fa/convolutional-neural-networks.md]: مصورکننده -> محصورکننده

File: fa/convolutional-neural-networks.md [hunk 18]
```diff
@@ -4,713 +4,919 @@
...
+</div>
 
 <br>
 
 
 **71. R-CNN ― Region with Convolutional Neural Networks (R-CNN) is an object detection algorithm that first segments the image to find potential relevant bounding boxes and then run the detection algorithm to find most probable objects in those bounding boxes.**
 
-&#10230;
+<div dir="rtl">
+R-CNN – ناحیه با شبکه‌های عصبی پیچشی (R-CNN) یک الگوریتم شناسایی شیء است که ابتدا تصویر را برای یافتن کادرهای مصورکننده مربوط بالقوه قطعهبندی می‌کند و سپس الگوریتم شناسایی را برای یافتن محتمل‌ترین اشیاء در این کادرهای محصور کننده اجرا می‌کند.
```
Comment (erfannoury) [inline fa/convolutional-neural-networks.md]: قطعه‌بندی

File: fa/convolutional-neural-networks.md [hunk 19]
```diff
@@ -4,713 +4,919 @@
...
+</div>
 
 <br>
 
 
 **75. Types of models ― Two main types of model are summed up in table below:**
 
-&#10230;
+<div dir="rtl">
+انواع مدل ها – دو نوع اصلی از مدل در جدول زیر بصورت خلاصه آورده‌شده‌اند:
```
Comment (erfannoury) [inline fa/convolutional-neural-networks.md]: انواع مدل
آورده شده‌اند

File: fa/convolutional-neural-networks.md [hunk 20]
```diff
@@ -4,713 +4,919 @@
...
+</div>
 
 <br>
 
 
 **77. [Is this the correct person?, One-to-one lookup, Is this one of the K persons in the database?, One-to-many lookup]**
 
-&#10230;
+<div dir="rtl">
+[فرد مورد نظر است؟، جستجوی یکبهیک، این فرد یکی از K فرد پایگاه داده است؟، جستجوی یکبهچند]
```
Comment (erfannoury) [inline fa/convolutional-neural-networks.md]: یک‌به‌یک
یک‌به‌چند

File: fa/convolutional-neural-networks.md [hunk 21]
```diff
@@ -4,713 +4,919 @@
...
+</div>
 
 <br>
 
 
 **78. One Shot Learning ― One Shot Learning is a face verification algorithm that uses a limited training set to learn a similarity function that quantifies how different two given images are. The similarity function applied to two images is often noted d(image 1,image 2).**
 
-&#10230;
+<div dir="rtl">
```
Comment (erfannoury) [inline fa/convolutional-neural-networks.md]: یک‌باره‌ای
اعمال‌شده بر روی
d(image 1, image 2)

File: fa/convolutional-neural-networks.md [hunk 22]
```diff
@@ -4,713 +4,919 @@
...
+</div>
 
 <br>
 
 
 **80. Triplet loss ― The triplet loss ℓ is a loss function computed on the embedding representation of a triplet of images A (anchor), P (positive) and N (negative). The anchor and the positive example belong to a same class, while the negative example to another one. By calling α∈R+ the margin parameter, this loss is defined as follows:**
 
-&#10230;
+<div dir="rtl">
+خطای سه‌گانه – خطای سه‌گانه ℓ یک تابع خطا است که بر روی بازنمایی تعبیه سه‌گانه تصاویر A (محور)، P (مثبت) و  N (منفی)  محاسبه می‌شود. نمونه‌های محور (anchor) و مثبت به دسته یکسانی تعلق دارند، حال آنکه نمونه منفی به دسته دیگری تعلق دارد. با نامیدن α∈R+ (به عنوان) فراسنج حاشیه، این خطا بصورت زیر تعریف می‌شود:
```
Comment (erfannoury) [inline fa/convolutional-neural-networks.md]: بازنمایی تعبیه‌ی سه‌گانه‌ی تصاویر

File: fa/convolutional-neural-networks.md [hunk 23]
```diff
@@ -4,713 +4,919 @@
...
+</div>
 
 <br>
 
 
 **83. [Content C, Style S, Generated image G]**
 
-&#10230;
+<div dir="rtl">
+[محتوا C، سبک S، تصویر تولیدشده G]
```
Comment (erfannoury) [inline fa/convolutional-neural-networks.md]: محتوای
تولید‌شده‌ی

File: fa/convolutional-neural-networks.md [hunk 24]
```diff
@@ -4,713 +4,919 @@
...
+</div>
 
 <br>
 
 
 **92. Generative Adversarial Network ― Generative adversarial networks, also known as GANs, are composed of a generative and a discriminative model, where the generative model aims at generating the most truthful output that will be fed into the discriminative which aims at differentiating the generated and true image.**
 
-&#10230;
+<div dir="rtl">
+GANs شناخته می‌شوند، ترکیبی از یک مدل مولد و تمیزدهنده هستند، جاییکه مدل مولد هدفش تولید واقعی‌ترین خروجی است که به (مدل) تمیزدهنده تغذیه می‌شود و این (مدل) هدفش تفکیک بین تصویر تولیدشده و واقعی است.
```
Comment (erfannoury) [inline fa/convolutional-neural-networks.md]: Translation of the first sentence is missing.
شبکه‌ی هم‌آورد مولد

File: fa/convolutional-neural-networks.md [hunk 25]
```diff
@@ -4,713 +4,919 @@
...
+</div>
 
 <br>
 
 
 **95. ResNet ― The Residual Network architecture (also called ResNet) uses residual blocks with a high number of layers meant to decrease the training error. The residual block has the following characterizing equation:**
 
-&#10230;
+<div dir="rtl">
+ResNet – معماری شبکه پسماند (همچنین با عنوان ResNet شناخته میشود) از بلاک‌های پسماند با تعداد لایه‌های زیاد به منظور کاهش خطای آموزش استفاده میکند. بلاک پسماند معادله‌ای با خصوصات زیر دارد:
```
Comment (erfannoury) [inline fa/convolutional-neural-networks.md]: خصوصات ->‌ خصوصیات

**Comment by AlisterTA** [comment]
@erfannoury thank you for taking your time reviewing this PR. (Also sorry if there was a ton of typos. I'm dyslexic and always  need someone to give me feedback )
@shervinea it's ready to be merged.

**Comment by erfannoury** [comment]
@AlisterTA no problem at all. I'll take a second look now.

**Comment by erfannoury** [comment]
LGTM. @shervinea can you please merge?

**Comment by shervinea** [comment]
Manually merged. Thank you for your work @erfannoury @AlisterTA @ehsanmk!

**Comment by erfannoury** [comment]
Thank you to @AlisterTA and @ehsanmk for their efforts in translating this document.
Thank you @shervinea.

**Comment by ehsanmk** [comment]
Thank you to @AlisterTA for correct my mistake and appreciate @erfannoury  for edit and review.

**Pull request closed without merging**